pandas/tseries/tests/test_timeseries.py

```diff
@@ -772,6 +772,48 @@ def test_to_datetime_unit(self):
         result = to_datetime([1, 2, 111111111], unit='D', errors='coerce')
         tm.assert_index_equal(result, expected)
 
+    def test_to_datetime_origin(self):
+        units = ['D', 's', 'ms', 'us', 'ns']
+        # Addresses Issue Number 11276, 11745
+        # for origin as julian
+        julian_dates = pd.date_range(
+            '2014-1-1', periods=10).to_julian_date().values
+        result = Series(pd.to_datetime(
+            julian_dates, unit='D', origin='julian'))
```

Comment:
You can omit the `Series( ..)` here and in the 'expected' below as well. It will then return indexes (so need to use `assert_index_equal` instead of series), but it will make the test a little bit easier to follow